src/dataflow/analysis/flow-graph.ts

```diff
@@ -59,10 +59,24 @@ export class FlowGraph {
     recipe.handleConnections.forEach(connection => {
       const particleNode = particleNodes.get(connection.particle);
       const handleNode = handleNodes.get(connection.handle);
-      const edgeId = 'E' + edgeIdCounter++;
-      const edge = addHandleConnection(particleNode, handleNode, connection, edgeId);
-      this.edges.push(edge);
-      this.handleSpecMap.set(connection.spec, edge);
+      this.handleSpecMap.set(connection.spec, handleNode);
+      
+      // Function to construct a new edge in the graph.
+      const addEdgeWithDirection = (direction: 'in' | 'out') => {
+        const edgeId = 'E' + edgeIdCounter++;
+        const edge = addHandleConnection(direction, particleNode, handleNode, connection, edgeId);
+        this.edges.push(edge);
+      };
+
+      if (connection.direction === 'inout') {
+        // An inout handle connection is represented by two edges.
+        addEdgeWithDirection('in');
+        addEdgeWithDirection('out');
+      } else if (connection.direction === 'in' || connection.direction === 'out') {
+        addEdgeWithDirection(connection.direction);
+      } else {
+        throw new Error(`Unsupported handle connection direction: ${connection.direction}`);
```

Comment:
Hadn't considered this before, but does this mean that graphs containing Slandles will throw?

Reply:
I'm not sure... We have support for slots (consume/provide) in line 83. I don't imagine it will be terribly hard to change that to work for slandles?